pkg/manifestgen/install/manifests.go

```diff
@@ -125,7 +126,12 @@ func build(base, output string) error {
 		return err
 	}
 
-	if err = os.WriteFile(output, resources, 0o640); err != nil {
+	outputBase := filepath.Dir(strings.TrimSuffix(output, string(filepath.Separator)))
+	fs, err := filesys.MakeFsOnDiskSecure(outputBase)
+	if err != nil {
+		return err
+	}
+	if err = fs.WriteFile(output, resources); err != nil {
```

Comment:
There are many places where we do this with `os.WriteFile`. Maybe we should do a followup PR and use our SecureFS everywhere.

Reply:
I would rather like to introduce our own https://pkg.go.dev/io/fs implementation which is consistently used, than to further build on Kustomize's FS interface.